src/live.rs

```diff
@@ -112,6 +114,49 @@ pub fn iso_ignition_remove(config: &IsoIgnitionRemoveConfig) -> Result<()> {
     Ok(())
 }
 
+pub fn pxe_ignition_wrap(config: &PxeIgnitionWrapConfig) -> Result<()> {
+    if config.output.is_none()
+        && isatty(stdout().as_raw_fd()).chain_err(|| "checking if stdout is a TTY")?
+    {
+        bail!("Refusing to write binary data to terminal");
+    }
+
+    let ignition = match config.ignition {
+        Some(ref ignition_path) => {
+            read(ignition_path).chain_err(|| format!("reading {}", ignition_path))?
+        }
+        None => {
+            let mut data = Vec::new();
+            stdin()
+                .read_to_end(&mut data)
+                .chain_err(|| "reading stdin")?;
+            data
+        }
+    };
+
+    let cpio = make_cpio(&ignition)?;
+
+    match &config.output {
+        Some(output_path) => {
+            write(output_path, cpio).chain_err(|| format!("writing {}", output_path))?
+        }
+        None => {
+            stdout().write_all(&cpio).chain_err(|| "writing output")?;
+            stdout().flush().chain_err(|| "flushing output")?;
+        }
+    }
+    Ok(())
+}
+
+pub fn pxe_ignition_unwrap(config: &PxeIgnitionUnwrapConfig) -> Result<()> {
+    let buf = read(&config.input).chain_err(|| format!("reading {}", config.input))?;
```

Comment:
Can we limit the size here to something reasonable? (I'm imagining someone passing the base initrd by accident instead of the Ignition cpio and we `malloc` 700M).

Reply:
Discussed OOB.  The large-file case is not great, but it's also not clear where to set a threshold, since people might encode large binaries into their Ignition configs.